core/transport/src/main/java/alluxio/util/grpc/GrpcChannel.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.util.grpc;
+
+import io.grpc.CallOptions;
+import io.grpc.Channel;
+import io.grpc.ClientCall;
+import io.grpc.MethodDescriptor;
+
+/**
+ * A simple wrapper around the {@link Channel} class in grpc. Outside of this module, this
+ * class should be used to replace references to {@link Channel} for dependency management.
```

Comment:
why is this useful for dependency management?

Reply:
the invariant is that there should be no references to `io.grpc` outside the transport module and we relocate `io.grpc` when building transport

Reply:
ah, that makes sense now. Do you know why we we chose to shade/relocate within only one project instead of across the uber-jar? It seems like this way has more boilerplate like this, and is also more error-prone since we can accidentally reference `io.grep` from outside this module by accident

Reply:
we're fine on the server side but the client side side was messy and had issues (don't remember exactly what messed up though). 

we had the same kind of issue with proto, and the solution of wrapping grpc classes is an extension of what we did there. in the present code, there are no proto references outside the protobuf module (now renamed to transport)